File: 2022 maj/python/2.py
with open("liczby.txt", "r") as f:
    file = list(map(int, f.read().strip().split('\n')))

ans1, ans2 = [0, 0], [0, 0]
for n in file:
    cnt, r, _n, i = 0, 0, n, 2
    while _n != 1:
        if _n % i == 0:
            r += 1
        
        while _n % i == 0:
            cnt += 1
            _n //= i

        i += 1

    if cnt > ans1[1]:
        ans1 = [n, cnt]
    if r > ans2[1]:
        ans2 = [n, r]
print(*ans1, *ans2)

File: 2022 maj/c++/2.cpp
#include <iostream>
#include <fstream>
using namespace std;


int main()
{
    ifstream in("liczby.txt");
    
    int n;
    pair <int, int> ans1 = {0, 0}, ans2 = {0, 0};
    while (in >> n)
    {
        int i = 2, cnt = 0, r = 0, _n = n;
        while (n != 1)
        {
            if (n % i == 0)
                r++;

            while (n % i == 0)
            {
                n /= i;
                cnt++;
            }
            i++;
        }
        if (cnt > ans1.second)
            ans1 = {_n, cnt};
        if (r > ans2.second)
            ans2 = {_n, r};
    }
    cout << ans1.first << ' ' << ans1.second << ' ' << ans2.first << ' ' << ans2.second << '\n';
}